src/main/java/seedu/address/storage/JsonAdaptedInterview.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.storage;
+
+import java.time.LocalDateTime;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.interview.Interview;
+import seedu.address.model.position.Position;
+
+public class JsonAdaptedInterview {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Interview's %s field is missing!";
+
+    private final JsonAdaptedApplicant applicant;
+    private final LocalDateTime date;
+    private final JsonAdaptedPosition position;
+
+    /**
+     * Constructs a {@code JsonAdaptedInterview} with the given interview details.
+     */
+    @JsonCreator
+    public JsonAdaptedInterview(@JsonProperty("applicant") JsonAdaptedApplicant applicant,
+            @JsonProperty("date") LocalDateTime date, @JsonProperty("position") JsonAdaptedPosition position) {
+        this.applicant = applicant;
+        this.date = date;
+        this.position = position;
+    }
+
+    /**
+     * Converts a given {@code Interview} into this class for Jackson use.
+     */
+    public JsonAdaptedInterview(Interview source) {
+        applicant = new JsonAdaptedApplicant(source.getApplicant());
+        date = source.getDate();
+        position = new JsonAdaptedPosition(source.getPosition());
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted applicant object into the model's {@code Interview} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted applicant.
+     */
+    public Interview toModelType() throws IllegalValueException {
+        final Applicant modelApplicant = applicant.toModelType();
+        final LocalDateTime modelDate = date;
```

Comment:
In their `JsonAdaptedPerson.java` they check for the data constraint violations here but we don't do any checking for our new attributes?